index.d.ts

```diff
@@ -1686,6 +1693,46 @@ declare function structuredClone(
   options?: ServiceWorkerGlobalScopeStructuredCloneOptions
 ): any;
 
+declare class urlURL {
+  constructor(url: string, base?: string);
+  readonly origin: string;
+  href: string;
+  protocol: string;
+  username: string;
+  password: string;
+  host: string;
+  hostname: string;
+  port: string;
+  pathname: string;
+  search: string;
+  hash: string;
+  readonly searchParams: urlURLSearchParams;
+  toJSON(): string;
+  toString(): string;
+}
+
+declare class urlURLSearchParams {
+  constructor(init?: urlURLSearchParamsInitializer);
+  append(name: string, value: string): void;
+  delete(name: string): void;
+  get(name: string): string | null;
+  getAll(name: string): string[];
+  has(name: string): boolean;
+  set(name: string, value: string): void;
+  sort(): void;
+  entries(): IterableIterator<string[]>;
+  keys(): IterableIterator<string>;
+  values(): IterableIterator<string>;
+  forEach(callback: Function, thisArg?: any): void;
+  toString(): string;
+  [Symbol.iterator](): IterableIterator<string[]>;
+}
+
+declare type urlURLSearchParamsInitializer =
+  | Iterable<Iterable<string>>
+  | Record<string, string>
+  | string;
+
```

Comment:
huh, I wonder what happened here. cc @vlovich

Reply:
We now have two URL implementations. This is the standards one. We're discussing internally how to manage this. Short term will be to override it so that standard one wins but we'll need to figure out a longer-term solution about how we manage types for different compatibility dates.